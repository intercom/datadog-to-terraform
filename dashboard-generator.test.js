import { generateDashboardTerraformCode } from "./dashboard-generator";
import screenboardJSON from "./examples/screenboard.json";
import timeboardJSON from "./examples/timeboard.json";

it("converts screenboards correctly", () => {
  expect(generateDashboardTerraformCode("sb_1", screenboardJSON)).toMatchSnapshot();
});

it("converts timeboards correctly", () => {
  expect(generateDashboardTerraformCode("tb_1", timeboardJSON)).toMatchSnapshot();
});
