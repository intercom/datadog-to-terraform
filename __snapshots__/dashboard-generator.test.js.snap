// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`converts screenboards correctly 1`] = `
"resource \\"datadog_dashboard\\" \\"sb_1\\" {
title = \\"Laura's Screenboard 7 Sep 2020 13:18\\"
description = \\"\\"
widget {
timeseries_definition {
request {
q = \\"avg:system.cpu.user{*}\\"
display_type = \\"line\\"
style = {
palette = \\"dog_classic\\"
line_type = \\"solid\\"
line_width = \\"normal\\"
}
}
yaxis = {
label = \\"\\"
scale = \\"linear\\"
min = \\"auto\\"
max = \\"auto\\"
include_zero = true
}
title = \\"Avg of system.cpu.user over *\\"
title_size = \\"16\\"
title_align = \\"left\\"
time = {
}
show_legend = false
}
layout = {
x = 17
y = 15
width = 47
height = 15
}
}
widget {
heatmap_definition {
request {
q = \\"avg:system.cpu.user{*}\\"
style = {
palette = \\"dog_classic\\"
}
}
title = \\"Avg of system.cpu.user over *\\"
title_size = \\"16\\"
title_align = \\"left\\"
show_legend = false
time = {
}
}
layout = {
x = 78
y = 7
width = 47
height = 15
}
}
widget {
query_value_definition {
request {
q = \\"avg:datadog.agent.running{*}\\"
aggregator = \\"avg\\"
conditional_formats {
comparator = \\"<=\\"
value = 1
palette = \\"white_on_green\\"
}
conditional_formats {
comparator = \\">\\"
value = 1
palette = \\"white_on_yellow\\"
}
conditional_formats {
comparator = \\">=\\"
value = 3
palette = \\"white_on_red\\"
}
}
title = \\"Avg of datadog.agent.running over *\\"
title_size = \\"16\\"
title_align = \\"left\\"
time = {
}
autoscale = true
precision = 2
}
layout = {
x = 8
y = 35
width = 47
height = 15
}
}
layout_type = \\"free\\"
is_read_only = false
notify_list = []
}"
`;

exports[`converts timeboards correctly 1`] = `
"resource \\"datadog_dashboard\\" \\"tb_1\\" {
title = \\"Laura's Timeboard 21 Aug 2020 16:30\\"
description = \\"\\"
widget {
timeseries_definition {
request {
q = \\"avg:system.cpu.user{*}\\"
display_type = \\"line\\"
style = {
palette = \\"dog_classic\\"
line_type = \\"solid\\"
line_width = \\"normal\\"
}
}
yaxis = {
label = \\"\\"
scale = \\"linear\\"
min = \\"auto\\"
max = \\"auto\\"
include_zero = true
}
title = \\"Avg of system.cpu.user over *\\"
show_legend = false
legend_size = \\"0\\"
}
}
widget {
query_value_definition {
request {
q = \\"avg:datadog.agent.running{*}\\"
aggregator = \\"avg\\"
conditional_formats {
comparator = \\"<\\"
value = 1
palette = \\"white_on_red\\"
}
conditional_formats {
comparator = \\">=\\"
value = 1
palette = \\"white_on_green\\"
}
}
title = \\"Avg of datadog.agent.running over *\\"
time = {
}
autoscale = true
precision = 2
}
}
template_variable {
name = \\"major\\"
default = \\"*\\"
prefix = \\"agent_version_major\\"
}
template_variable {
name = \\"minor\\"
default = \\"*\\"
prefix = \\"agent_version_minor\\"
}
template_variable {
name = \\"patch\\"
default = \\"*\\"
prefix = \\"agent_version_patch\\"
}
layout_type = \\"ordered\\"
is_read_only = false
notify_list = []
template_variable_preset {
name = \\">= 7\\"
template_variable {
name = \\"major\\"
value = \\"7\\"
}
}
template_variable_preset {
name = \\"Latest\\"
template_variable {
name = \\"major\\"
value = \\"7\\"
}
template_variable {
name = \\"minor\\"
value = \\"21\\"
}
template_variable {
name = \\"patch\\"
value = \\"1\\"
}
}
}"
`;
